### PR TITLE
You Are Two mutation

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Makes you two.

You have two bodies, they both do what you're doing, or at least attempt to.

You have an ability to swap focus on which body, which lets you do some fine manipulation I can't replicate with signals.

### Advantages of this:
If you can keep them close together (try walking into walls), you basically get to attack with two people
One can "save" the other
Equip em both with stuff, you have two fellas worth of space
Be in two places at once... I mean, slime split is probably better for this, but what do I know?
### Disadvantages of this:
Both have to stay alive
Where the fuck is the other one
Both have to stay alive
Controlling two people at the same time
Both have to stay alive
Dear god

- [ ] hook more signals for attacking, and other stuff
- [x] make one dying kill the other
- [x] add an achievement for escaping with two
- [ ] add some sanity for not getting another "you are two" mutation on the other
- [x] ...copy some more stuff over like mutations and traumas? they should actually be copies, or else this is just hars heal trick again
- [ ] so a solution for the above was devised, BUT: i must refactor mutations and traumas to split behavior on "first applied" and "enabled" as to not have weird stacking/repeating effects that shouldn't when i transfer a mutation to a new person without exploits. same for traumas, most likely. maybe make a proc for this?
- [x] implement a way to prevent overlapping... you know, to make it more interesting.

p.s. I would have hooked more signals before pr'ing this but something came up and I wanna farm upvotes while i'm away. #honesty

## Why It's Good For The Game

This seems like an incredibly fun concept for a mutation. Also, that achievement has GOT to happen. Who is up for a challenge?

## Changelog
:cl:
add: New mutation: You Are Two!
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
@tralezab You litte F**ker
You made a shit of piece with your trash SS13 it’s f**King Bad this trash game I will become back my money I hope you will in your next time a cow on a trash farm you sucker

**Comment by Fikou** [comment]
@tralezab You litte F##ker
You made a shit of piece with your trash SS13 it’s f##King Bad this trash game I will become back my money I hope you will in your next time a cow on a trash farm you sucker

File: code/datums/mutations/you_are_two.dm [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+/obj/effect/proc_holder/spell/self/you_are_the_other
+	name = "\"You are the other\""
+	desc = "Moves your main focus to your other self, for more fine tuned interactions."
+	charge_max = 50
+	cooldown_min = 50
+	clothes_req = FALSE
+	action_icon_state = "declaration"
+
+/obj/effect/proc_holder/spell/pointed/declare_evil/cast(list/targets, mob/living/carbon/human/user, silent = FALSE)
```
Comment (Fikou) [inline code/datums/mutations/you_are_two.dm:61]: copy pastea

**Comment by NoxVS** [comment]
How does this interact with one or both being a slimeperson?

**Comment by tralezab** [comment]
Mmm... probably not well. I should make a mind transfer blocked trait. Or try it out in game first

**Comment by ShibaInuLord** [comment]
i cant believe jacob and esau are in ss13 now

**Comment by maxymax13** [comment]
if you speak into common does it show up twice 🤔

**Comment by tralezab** [comment]
Yes

**Comment by D4C-420** [comment]
It’s all fun and games until someone makes a tainted you are two mutation.

**Comment by tralezab** [comment]
this is great
image

**Comment by TemporalOroboros** [comment]
This doesn't look like the other gets the mutations of the original. Could that let you stack this basically infinitely?

**Comment by tralezab** [comment]
I'm still thinking of a good solution for that. Probably going to involve simply making the other immutable.

**Comment by optimumtact** [comment]
this is going to be buggier than dullahans

**Comment by tralezab** [comment]
Honestly if dullahans used signals like I am they wouldn't be buggy, so I have faith on making this work

File: code/datums/mutations/you_are_two.dm [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+/datum/mutation/human/you_are_two/proc/revive_one_of_them(mob/living/reviving)
+	var/genetic_punishment = the_other.health < HEALTH_THRESHOLD_DEAD
+	reviving.revive(full_heal = genetic_punishment)
+	if(genetic_punishment)//we need the person in question revived at all costs so we'll apply some really annoying damage type to encourage fixing both bodies
+		reviving.visible_message("<span class='notice'>[src]'s wounds close, leaving severe genetic imperfections in their place!</span>")
+		reviving.adjustCloneLoss(70)
+
+///called when the one moves
+/datum/mutation/human/you_are_two/proc/on_pre_move(datum/source, atom/newloc)
```
Comment (tralezab) [inline code/datums/mutations/you_are_two.dm:82]: note to self: not deep enough, still checking if movement is possible

**Comment by optimumtact** [comment]
> Honestly if dullahans used signals like I am they wouldn't be buggy, so I have faith on making this work

Laughing newspaper editor from spiderman movies meme macro

**Comment by tralezab** [comment]
Well... If I keep it modular, may we at least try? If I pull this off, this will be pretty fucking rad

**Comment by D4C-420** [comment]
Oh yeah how does this work with mutadone. If one gets gibbed or dusted do they both? For revival do they both have even be in the original body? How is health displayed to the player. Inventory as well.  . Do rev heads who have a clone off and on station count as on? Do spells/ gene/ ling/ body augments get carried over to the clone? going on here but these are just some interactions I thought up.

**Comment by tralezab** [comment]
>Oh yeah how does this work with mutadone.

copy will be immutable, original will lose the mutation and lose their copy. Hell, maybe I should make the mutations pass between humans when you switch... interesting thought, i may pursue!

>If one gets gibbed or dusted do they both?

Yes!
>For revival do they both have even be in the original body?

For things that revive you, you must be in your body. one will revive when the other does.
>How is health displayed to the player. Inventory as well.

You have an ability to swap to the other body (making your original copy your movements) to do fine tuning like that
>Do rev heads who have a clone off and on station count as on?

The mind will follow the active one, making the rev considered as abandoning if it is actively using the off station clone. Either useless or forfeit, but either way they both die when the one onstation gets killed so i'm not worried!
>Do spells/ gene/ ling/ body augments get carried over to the clone?

The big deliberation i've been having. Yeah, i'll make mind specific stuff carry over. I know spells already do, but lets also do genes. Lings will follow automatically. body augments won't.

>going on here but these are just some interactions I thought up.

it actually helps to think of edge cases here so thank you

**Comment by MMMiracles** [comment]
i can't wait to play my favorite binding of isaac character, jacob & esau

**Comment by Wallemations** [comment]
IMAGINATION GREENTEXT!!
image0

**Comment by tralezab** [comment]
So let me just say right now

This is going better than I thought it would

**Comment by OceanFish1** [comment]
This is great. 
If it works I won't become back my money

**Comment by Colovorat** [comment]
Reintroduce dullahans Pog

**Comment by Farquaar** [comment]
> 
> 
> this is great
> image
1. Give yourself the mutation
2. Line up the HoPline
3. Waste the HoP's time and clog up the line
4. Profit?

**Comment by tralezab** [comment]
@Farquaar 
I've been messing around with the double attacks, and they're really great.

**Comment by Qustinnus** [comment]
Dullahans were made by someone who sucks ass at coding so there's a bit of a difference there

File: code/datums/achievements/misc_achievements.dm [hunk 3]
```diff
@@ -171,3 +171,9 @@
 	desc = "Lights out, guerilla radio!"
 	database_id = MEDAL_TRAM_SURFER
 	icon = "tram_surfer"
+
+/datum/award/achievement/misc/double_escape
+	name = "Double Escape"
+	desc = "Escape alive with the \"You are Two\" mutation. Not so easy now, is it?"
```
Comment (Time-Green) [inline code/datums/achievements/misc_achievements.dm:177]: It is if you take the mutation three seconds before arriving on centcom. Maybe add a minimum time aswell?

**Comment by InsaneRed** [comment]
how is this gonna work with radstorm / monkeys

**Comment by tralezab** [comment]
All mutations that happen to the other are transferred to the one, and whenever the one switches to the other, the mutations follow the one.

**Comment by InsaneRed** [comment]
well if you let monkeys have this wont it allow xenobio to infinetly farm slimes?

**Comment by tralezab** [comment]
Yeah I can remove the mutation on monkify, if it doesn't do that already. Can... monkeys use all mutations?

File: code/datums/achievements/misc_achievements.dm (hunk 3 above)
Comment (tralezab) [reply code/datums/achievements/misc_achievements.dm:177]: sure

File: .vscode/launch.json [hunk 4]
```diff
@@ -6,7 +6,8 @@
       "request": "launch",
       "name": "Launch DreamSeeker",
       "preLaunchTask": "Build All",
-      "dmb": "${workspaceFolder}/${command:CurrentDMB}"
+      "dmb": "${workspaceFolder}/${command:CurrentDMB}",
+        "dreamDaemon": true
```
Comment (tralezab) [inline .vscode/launch.json:10]: i'll remove this, but this should really be a default setting

**Comment by InsaneRed** [comment]
> Yeah I can remove the mutation on monkify, if it doesn't do that already. Can... monkeys use all mutations?

i dont think monkeys can use it on themselves but i meant mroe as in just get 1 monkey, twoifiey it and then kill the second one and claim it with xeno machine to get it stored on the computer and repeat

**Comment by tralezab** [comment]
@InsaneRed I'm pretty sure there are easier ways to get monkeys that that, but the other part is that if one of the two is gibbed the other one is too... or should be!

**Comment by tralezab** [comment]
heh. heheh. Ybeert Ewolram,

image

**Comment by moocowswag** [comment]
I can see one really buggy / anomaly that might form when a splitty slime person with this mutation activated splits.

Activated mutations are also activated on new bodies when a slime person splits so you could end up with double the bodies on each split


Also if you are controlling both bodies at the same time how does that work with spells / mutations that provide an active ability.

Take firebreath for example, will both bodies fire a fireball at the same time or will the one youre controlling fire, and if only one of them fires will both bodies go on cooldown for firebreath or only one of them?

**Comment by tralezab** [comment]
It is possible to lock mutations to species, I could simply lock them to roundstart species only and that'd go a long way towards preventing weird stuff with other species. Which would also fix the monkey issue.

**Comment by SmArtKar** [comment]
Hm, how will this work with changelings? I suspect that changeling's revival might be broken with them

**Comment by tralezab** [comment]
Should work as intended. One reviving revives the other.

**Comment by tralezab** [comment]
this was genuinely fun to work on, but there's /just/ too many variables to consider. The other body basically needs to not be able to be possessed, and into anything inhuman, be duplicated, etc etc etc. I think I could get it to be stable enough given enough time, but

**Pull request closed without merging**